### ISSUE TITLE
trailing negator is not caught

### ISSUE THREAD

**Issue by poseidonCore** [Microsoft/TypeScript#28469]
**TypeScript Version:**  3.1 TS playground @ 11 NOV 2018
 
TypeScript is being over-generous in allowing a hanging negator that should really be highlighted as a syntax error, despite the fact that it transpiles away the error in the JS version.

**Priority**: low

**Search Terms:**
Hanging negator / negation operator.

**Code**

```ts
var answer = true! || true;
```

**Expected behavior:**
var answer = true! || true;
 .................................^
SyntaxError: Unexpected token !

**Actual behavior:**
Transpiles to ``var answer = true || true;``

**Playground Link:** https://www.typescriptlang.org/play/index.html#src=var%20answer%20%3D%20true!%20%7C%7C%20true%3B%0D%0A

**Related Issues:**

**Comment by yortus** [comment]
That's not a negator, it's TypeScript's [non-null assertion operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator).
